crates/storage/db/src/tables/models/mod.rs

```diff
@@ -135,3 +136,20 @@ impl Decode for StoredNibblesSubKey {
         Ok(Self::from_compact(buf, buf.len()).0)
     }
 }
+
+impl Encode for PrunePart {
+    type Encoded = [u8; 1];
+
+    fn encode(self) -> Self::Encoded {
+        let mut bytes = BytesMut::with_capacity(1);
+        self.to_compact(&mut bytes);
+        bytes.as_ref().try_into().expect("incorrect length")
+    }
```

Comment:
@joshieDo there surely is a better way to do that but I can't figure that out.

The most obvious way
```rust
let mut buf = [0u8];
self.to_compact(&mut buf);
buf
```
doesn't work:
```console
error[E0277]: the trait bound `[u8; 1]: BufMut` is not satisfied
   --> crates/storage/db/src/tables/models/mod.rs:145:25
    |
145 |         self.to_compact(&mut buf);
    |              ---------- ^^^^^^^^ the trait `BufMut` is not implemented for `[u8; 1]`
    |              |
    |              required by a bound introduced by this call
    |
    = help: the trait `BufMut` is implemented for `&mut [u8]`
note: required by a bound in `to_compact`
   --> /Users/shekhirin/Projects/reth/crates/storage/codecs/src/lib.rs:32:12
    |
32  |         B: bytes::BufMut + AsMut<[u8]>;
    |            ^^^^^^^^^^^^^ required by this bound in `Compact::to_compact`
help: convert the array to a `&mut [u8]` slice instead
    |
145 |         self.to_compact(&mut buf[..]);
    |                                 ++++
```

Following the compiler suggestion and doing 
```rust
let mut buf = [0u8];
self.to_compact(&mut buf[..]);
buf
```
also doesn't work:
```console
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> crates/storage/db/src/tables/models/mod.rs:145:25
    |
145 |         self.to_compact(&mut buf[..]);
    |              ---------- ^^^^^^^^^^^^ doesn't have a size known at compile-time
    |              |
    |              required by a bound introduced by this call
    |
    = help: the trait `Sized` is not implemented for `[u8]`
note: required by a bound in `to_compact`
   --> /Users/shekhirin/Projects/reth/crates/storage/codecs/src/lib.rs:30:19
    |
30  |     fn to_compact<B>(self, buf: &mut B) -> usize
    |                   ^ required by this bound in `Compact::to_compact`
```

---

I suspect there's something in how we restrict the `B` in `to_compact` but don't understand straight away.

Reply:
solved